src/main/java/gov/usgs/earthquake/indexer/ReliableIndexerListener.java

```diff
@@ -0,0 +1,258 @@
+
+      List<ProductSummary> productList = null;
+
+      //Synchronized so we aren't notified of new products right before we wait
+      synchronized (syncObject) {
+        try {
+          productList = getNextProducts();
+        } catch (Exception e) {
+          //Not a phenomenal way to handle this exception
+          LOGGER.log(Level.WARNING, "[" + getName() + "] Exception getting next products", e);
+        }
+        if (productList == null || productList.size() == 0) {
+          try {
+            //Wait when there are no more products to process
+            syncObject.wait();
+          } catch (InterruptedException ignore) {
+            //Ignore because it's most likely we get interrupted by shutdown
+            LOGGER.log(Level.FINE,"[" + getName() + "] was told to stop, or something went wrong");
+          }
+          continue;
+        }
+      }
+
+      //Process the products we have
+      for(ProductSummary summary : productList) {
+        LOGGER.log(Level.FINEST,"[" + getName() + "] preparing to process product " + summary.getIndexId());
+        //Check for shutdown every iteration so we don't hog shutdown time
+        if (stopThread) {
+          break;
+        }
+        try {
+          //Process the product types we're told to in configuration
+          LOGGER.log(Level.FINEST,"[" + getName() + "] determining if we can process product " + summary.getIndexId());
+          if (accept(summary.getId())) {
+            LOGGER.log(Level.FINEST,"[" + getName() + "] attempting to process product " + summary.getIndexId());
+            this.processProduct(summary);
+          }
+          //Update internal storage so we don't reprocess products
+          this.setLastIndexId(summary.getIndexId());
+        } catch (Exception e) {
+          try {
+            //Handle exception if we can
+            this.onProcessException(summary,e);
+          } catch(Exception e2) {
+            //Give up if we can't
+            break;
+          }
+        }
+      }
+
+    }
+  }
+
+  /**
+   * Starts thread
+   * 
+   * Calls onBeforeProcessThreadStart() in case subclasses want to add functionality
+   *
+   * @throws Exception if there's a thread issue
+   * @throws Exception if thread start fails
+   */
+  @Override
+  public void startup() throws Exception{
+    super.startup();
+    this.onBeforeProcessThreadStart();
+    this.processThread = new Thread(this);
+    this.processThread.start();
+  }
+
+  /**
+   * Closes thread
+   *
+   * @throws Exception if there's a thread issue
+   */
+  @Override
+  public void shutdown() throws Exception {
+    try {
+      LOGGER.log(Level.FINEST,"[" + getName() + "] trying to shut down...");
+      stopThread = true;
+      //When the thread is ready, tell it to stop
+      synchronized (syncObject) {
+        this.processThread.interrupt();
+      }
+      this.processThread.join();
+    } finally {
+      super.shutdown();
+    }
+  }
+
+  @Override
+  public String getName() {
```

Comment:
Don't override this method, it's configured during Config loading...

Reply:
I was overriding it because the "names" were always null on the ouput log. Is there something I should do to fix that